cross/libaom/Makefile

```diff
@@ -0,0 +1,96 @@
+INSTALL_TARGET = libaom_custom_install
+
+NASM_PATH = $(WORK_DIR)/../../../native/nasm/work-native/install/usr/local/bin
+
+CMAKE_ARGS = -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX)
+CMAKE_ARGS += -DCMAKE_CROSSCOMPILING=TRUE
+CMAKE_ARGS += -DCMAKE_SYSTEM_NAME=Linux
+CMAKE_ARGS += -DCMAKE_C_COMPILER=$(TC_PATH)$(TC_PREFIX)gcc
+CMAKE_ARGS += -DCMAKE_CXX_COMPILER=$(TC_PATH)$(TC_PREFIX)g++
+CMAKE_ARGS += -DCMAKE_INSTALL_RPATH=$(INSTALL_PREFIX)/lib
+CMAKE_ARGS += -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
+CMAKE_ARGS += -DCMAKE_BUILD_WITH_INSTALL_RPATH=TRUE
+CMAKE_ARGS += -DCONFIG_AV1_ENCODER=1 -DBUILD_SHARED_LIBS=1
+
+include ../../mk/spksrc.cross-cc.mk
+
+# Set working directory
+PKG_WORK_DIR = $(WORK_DIR)/$(PKG_NAME)_$(PKG_VERS)-build
+
+ifeq ($(findstring $(ARCH), $(ARM5_ARCHES)),$(ARCH))
+CMAKE_ARGS += -DCROSS_COMPILE_ARM=ON -DENABLE_ASSEMBLY=OFF
+CMAKE_ARGS += -DCMAKE_SYSTEM_PROCESSOR=armv5 -DAOM_TARGET_CPU=arm
+endif
+
+ifeq ($(findstring $(ARCH), $(ARM7_ARCHES)),$(ARCH))
+CMAKE_ARGS += -DCMAKE_CXX_FLAGS=-fPIC -DCROSS_COMPILE_ARM=ON -DENABLE_ASSEMBLY=OFF
+CMAKE_ARGS += -DCMAKE_SYSTEM_PROCESSOR=armv7 -DAOM_TARGET_CPU=arm
+#CMAKE_ARGS += -DCMAKE_TOOLCHAIN_FILE=$(WORK_DIR)/$(PKG_DIR)/build/cmake/toolchains/armv7-linux-gcc.cmake
+endif
+
+ifeq ($(findstring $(ARCH), $(ARM8_ARCHES)),$(ARCH))
+CMAKE_ARGS += -DCMAKE_CXX_FLAGS=-fPIC -DCROSS_COMPILE_ARM=ON -DENABLE_ASSEMBLY=OFF
+CMAKE_ARGS += -DCMAKE_SYSTEM_PROCESSOR=aarch64 -DAOM_TARGET_CPU=arm64
+#CMAKE_ARGS += -DCMAKE_TOOLCHAIN_FILE=$(WORK_DIR)/$(PKG_DIR)/build/cmake/toolchains/arm64-linux-gcc.cmake
+endif
+
+ifeq ($(findstring $(ARCH), $(PPC_ARCHES)),$(ARCH))
+CMAKE_ARGS += -DCMAKE_C_FLAGS=-maltivec -DENABLE_ALTIVEC=OFF -DCPU_POWER8=OFF
+CMAKE_ARGS += -DCMAKE_SYSTEM_PROCESSOR=ppc64 -DAOM_TARGET_CPU=ppc
+endif
+
+ifeq ($(findstring $(ARCH), $(x64_ARCHES)),$(ARCH))
+DEPENDS += native/nasm cross/libnuma
+ENV += PATH=$(NASM_PATH):$$PATH
+ENV += AS=$(NASM_PATH)/nasm
+CMAKE_ARGS += -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DAOM_TARGET_CPU=x86_64
+#CMAKE_ARGS += -DCMAKE_TOOLCHAIN_FILE=$(WORK_DIR)/$(PKG_DIR)/build/cmake/toolchains/x86-linux.cmake
+endif
+
+ifeq ($(findstring $(ARCH), $(x86_ARCHES)),$(ARCH))
+DEPENDS += native/nasm cross/libnuma
+ENV += PATH=$(NASM_PATH):$$PATH
+ENV += AS=$(NASM_PATH)/nasm
+CMAKE_ARGS += -DENABLE_ASSEMBLY=OFF -DCMAKE_SYSTEM_PROCESSOR=x86
+CMAKE_ARGS += -DCMAKE_SYSTEM_PROCESSOR=x86 -DAOM_TARGET_CPU=x86
+#CMAKE_ARGS += -DCMAKE_TOOLCHAIN_FILE=$(WORK_DIR)/$(PKG_DIR)/build/cmake/toolchains/x86-linux.cmake
+endif
+
+.PHONY: libaom_custom_configure
+
+libaom_custom_configure: 
+	$(RUN) rm -rf CMakeCache.txt CMakeFiles
+	$(RUN) mkdir --parents $(PKG_WORK_DIR)
+	cd $(PKG_WORK_DIR) && env $(ENV) cmake $(CMAKE_ARGS) $(WORK_DIR)/$(PKG_DIR)
+
+.PHONY: libaom_custom_compile
+
+libaom_custom_compile: 
+	cd $(PKG_WORK_DIR) && env $(ENV) $(MAKE)
+
+.PHONY: libaom_custom_install
+
+libaom_custom_install: 
+	cd $(PKG_WORK_DIR) && env $(ENV) $(MAKE) install DESTDIR=$(INSTALL_DIR)
```

Comment:
That is the first time CMake sources are built that way in a dedicated build path.
Until now, sources are built "in place" in current source directory as a dedicated one is created for each architecture.
So the question: is it really worth to build "elsewhere" as our source directory cannot be considered as "developer workspace" but only temporary discarded compilation tree...

Reply:
First time for me as well.  I had to play with it quite a bit to have it working as libaom doesn't support in-directory building:
```
CMake Error at CMakeLists.txt:27 (message):
  Building from within the aom source tree is not supported.
```
Which explains why I ended-up building "elsewhere" ... Alternative ideas on this welcomed.

Reply:
Noticed. I will create a support for that in framework... one day...